### ISSUE TITLE
degree of number that is hidden zero

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#13050]
I'm not sure why it is important that `degree(Poly(0, x), x)` be `-oo` but `degree(Poly(1, x), x)` be 0 but I suppose there is some rationale. If there is, then more care should be taken to handle cases where a number is not known to be 0 or not:

```
>>> p = cos(1)**2+sin(1)**2-1
>>> Poly(p, x).degree()
0
>>> Poly(0, x).degree()
-oo
```

Perhaps this should be returned instead: `Piecewise((-oo, Eq(-1 + cos(1)**2 + sin(1)**2, 0)), (0, True))`

**Comment by asmeurer** [comment]
This is a general issue in polys. You can have the leading coefficient be actually 0, and it will produce wrong results.

**Comment by smichr** [comment]
> it will produce wrong results.

do you mean something like this:

```
>>> zero=cos(1)**2+sin(1)**2-1
>>> degree(zero*x**2+x)
2
```

Also, is it important that -oo be returned or could a simple -1 -- like that returned by 'abc'.find('z') -- suffice?

**Comment by asmeurer** [comment]
-oo is generally preferred for the degree of 0 as it works nicer with properties of the degree. For instance, `deg(p*q) = deg(p) + deg(q)` only works if `deg(0) = -oo`.

**Comment by asmeurer** [comment]
> do you mean something like this:

Yes, but it gets worse, because the division algorithm relies on detecting if coefficients are zero. Which in turn propagates to other things:

```
>>> Poly(x**2 - x, x).gcd(Poly(x + zero, x))
Poly(1, x, domain='ZZ[zero]')
>>> Poly(x**2 - x, x).gcd(Poly(x, x))
Poly(x, x, domain='ZZ')
```

It's very similar to the RREF algorithm for matrices, which can also be fooled by false zeros.
